src/term/check/set_entrypoint.rs

```diff
@@ -23,13 +22,17 @@ impl Display for EntryErr {
         write!(f, "File has '{}', '{}' and '{}' definitions.", fnd[0], fnd[1], fnd[2])
       }
       EntryErr::MultipleRules => write!(f, "Main definition can't have more than one rule."),
-      EntryErr::Arguments => write!(f, "Main definition can't have any arguments."),
     }
   }
 }
 
 impl Ctx<'_> {
   pub fn set_entrypoint(&mut self) {
+    // already setted
+    if self.book.entrypoint.is_some() {
```

Comment:
I only added this line because it will fail in the run mode.
Also, I don't see any reason to check for Main or main when there is an custom entrypoint.

Reply:
Why would it fail?

Reply:
The logic is wrong, it shows me 'File has both 'Main' and 'Main' definitions.'